### DISCUSSION TITLE
What causes Latex Workshop to recompile the same pdf multiple times?

### DISCUSSION THREAD

**Discussion by kaba2** [James-Yu/LaTeX-Workshop#4204] [Q&A]
I have a large latex file which Latex Workshop for some reason recompiles 4 times in succession.  What causes this? Supposedly the first build already produces a pdf file, which is then immediately invalidated by the next run. I wish I could just preview this first version, even if it is not quite right.

The runs I see in output are:

``` 
Run number 1 of rule 'pdflatex'
...
Run number 1 of rule 'biber output/dimension-paper'
...
Run number 2 of rule 'pdflatex'
...
Run number 3 of rule 'pdflatex'
```

I wrapped `\ usepackage{biblatex}`, `\printindex` and `\printbibliography` under a conditional flag. With this flag disabled, I can get rid of the biber run. However, the three other `pdflatex` remain.

**Comment by jlelong** [comment]
LW uses `latexmk` to build a project, which runs `pdflatex` sufficiently many times to get the references right. If you want a single compilation, define a new recipe that only calls `pdflatex` only once, see https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#latex-recipes
